### PR TITLE
Add "Join Table" for visits

### PR DESCRIPTION
**Pull request by braincee**
_**In this pull request:**_
- I created a table named vets with the following columns:
  - id: integer (set it as autoincremented PRIMARY KEY)
  - name: string
  - age: integer
  - date_of_graduation: date
- There is a many-to-many relationship between the tables `species` and `vets`: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called `specializations` to handle this relationship.
- There is a many-to-many relationship between the tables `animals` and `vets`: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called `visits` to handle this relationship, it should also keep track of the date of the visit.

- I inserted the following data for vets:
  - Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
  - Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
  - Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
  - Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
- I inserted the following data for specialties:
  - Vet William Tatcher is specialized in Pokemon.
  - Vet Stephanie Mendez is specialized in Digimon and Pokemon.
  - Vet Jack Harkness is specialized in Digimon.
 
- I inserted the following data for visits:
  - Agumon visited William Tatcher on May 24th, 2020.
  - Agumon visited Stephanie Mendez on Jul 22th, 2020.
  - Gabumon visited Jack Harkness on Feb 2nd, 2021.
  - Pikachu visited Maisy Smith on Jan 5th, 2020.
  - Pikachu visited Maisy Smith on Mar 8th, 2020.
  - Pikachu visited Maisy Smith on May 14th, 2020.
  - Devimon visited Stephanie Mendez on May 4th, 2021.
  - Charmander visited Jack Harkness on Feb 24th, 2021.
  - Plantmon visited Maisy Smith on Dec 21st, 2019.
  - Plantmon visited William Tatcher on Aug 10th, 2020.
  - Plantmon visited Maisy Smith on Apr 7th, 2021.
  - Squirtle visited Stephanie Mendez on Sep 29th, 2019.
  - Angemon visited Jack Harkness on Oct 3rd, 2020.
  - Angemon visited Jack Harkness on Nov 4th, 2020.
  - Boarmon visited Maisy Smith on Jan 24th, 2019.
  - Boarmon visited Maisy Smith on May 15th, 2019.
  - Boarmon visited Maisy Smith on Feb 27th, 2020.
  - Boarmon visited Maisy Smith on Aug 3rd, 2020.
  - Blossom visited Stephanie Mendez on May 24th, 2020.
  - Blossom visited William Tatcher on Jan 11th, 2021.

- I wrote queries to answer the following:
  - Who was the last animal seen by William Tatcher?
  - How many different animals did Stephanie Mendez see?
  - List all vets and their specialties, including vets with no specialties.
  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
  
Screenshot (158)

   - What animal has the most visits to vets?
   - Who was Maisy Smith's first visit?
   - Details for most recent visit: animal information, vet information, and date of visit.
   - How many visits were with a vet that did not specialize in that animal's species?

Screenshot (159)

   - What specialty should Maisy Smith consider getting? Look for the species she gets the most.
   
Screenshot (160)

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -15,3 +15,139 @@ SELECT * FROM animals WHERE neutered = True;
...
+-- What animal has the most visits to vets?
+
+SELECT animals.name AS animal_name, COUNT(animals.id) AS most_visits FROM visits
+INNER JOIN animals ON visits.animal_id = animals.id GROUP BY animals.name ORDER BY COUNT(animals.id) DESC LIMIT 1;
+
+-- Who was Maisy Smith's first visit?
+
+SELECT vets.name AS vets_name, visits.date_of_visit AS date_of_visit FROM vets
+INNER JOIN visits ON vets.id = visits.vet_id WHERE vets.name = 'Maisy Smith' ORDER BY visits.date_of_visit
+LIMIT 1;
```
Comment (AnselemOdims) [inline queries.sql:128]: - [ ] Please kindly include the animal name in the select column as the query asks `who`

File: queries.sql [hunk 2]
```diff
@@ -15,3 +15,139 @@ SELECT * FROM animals WHERE neutered = True;
...
+vets.age AS vets_age, vets.date_of_graduation AS date_of_graduation, visits.date_of_visit AS date_of_visit FROM visits 
+INNER JOIN animals ON visits.animal_id = animals.id INNER JOIN species ON animals.species_id = species.id
+INNER JOIN vets ON visits.vet_id = vets.id ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+
+SELECT vets.name AS vets_name, COUNT(visits.animal_id) AS number_of_visits,
+specializations.species_id AS species_specialities FROM visits JOIN animals ON animals.id = visits.animal_id FULL JOIN specializations 
+ON visits.vet_id = specializations.vet_id JOIN vets ON visits.vet_id = vets.id GROUP BY visits.animal_id, visits.vet_id, animals.name,
+specializations.species_id, vets.name ORDER BY COUNT(visits.animal_id) DESC LIMIT 1;
```
Comment (AnselemOdims) [inline queries.sql]: - [ ] This query does not return the visits that were with a vet that DID NOT specialize in animal's species. Please kindly make your query return the right results.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @braincee , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: queries.sql (hunk 2 above)
Comment (braincee) [reply queries.sql]: - I made changes that return the visits that were with a vet that DID NOT specialize in animal species.

Screenshot (165)
.

Comment (Whoistolu) [reply queries.sql]: Well resolved 👍🏽

File: queries.sql (hunk 1 above)
Comment (Whoistolu) [reply queries.sql:128]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @braincee 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by braincee** [comment]
Thanks very much, @AnselemOdims @Whoistolu 

thank-you-thanks

**Pull request merged**